File: resource-registry/src/main/java/nl/haploid/octowight/registry/repository/ResourceElementDmoRepository.java
package nl.haploid.octowight.registry.repository;

import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface ResourceElementDmoRepository extends MongoRepository<ResourceElementDmo, String> {

	ResourceElementDmo findByAtomIdAndAtomTypeAndAtomOrigin
			(final Long atomId, final String atomType, final String atomOrigin);

	List<ResourceElementDmo> findByAtomIdInAndAtomTypeAndAtomOrigin
			(final List<Long> atomId, final String atomType, final String atomOrigin);

	Long deleteByResourceTypeAndResourceId(final String resourceType, final long resourceId);
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/repository/ResourceElementDmoRepository.scala
package nl.haploid.octowight.registry.repository

import java.{lang, util}

import org.springframework.data.mongodb.repository.MongoRepository

trait ResourceElementDmoRepository extends MongoRepository[ResourceElementDmo, String] {

  def findByAtomIdAndAtomCategoryAndAtomOrigin(atomId: lang.Long, atomCategory: String, atomOrigin: String): ResourceElementDmo

  def findByAtomIdInAndAtomCategoryAndAtomOrigin(atomId: lang.Iterable[lang.Long], atomCategory: String, atomOrigin: String): util.List[ResourceElementDmo]

  def deleteByResourceTypeAndResourceId(resourceType: String, resourceId: lang.Long): Long
}
